### PR TITLE
Add abstracted Find Files functionality

### PR DESCRIPTION
**Pull request by nigurr**
- Multiple filters can be passed and should be sent as array
- Include/Exclude filter support
- Search files anywhere

### CODE REVIEW

File: node/task.ts [hunk 1]
```diff
@@ -1157,6 +1159,27 @@ export function globFirst(pattern: string): string {
     return matches[0];
 }
 
+export function findFiles(includePattern: string[], excludePattern: string[], workingDir?: string): string[] {
```
Comment (ericsciple) [inline node/task.ts]: does this use minimatch pattern matching rules?

Comment (ericsciple) [inline node/task.ts]: If exclude patterns is to be pulled into the lib, then tl.match should have an overload. At that point, this function would become a thin wrapper over tl.find and tl.match.

I like the idea of a wrapper function so caller doesn't need to setup the options (specify matchbase:true and if-windows case-insensitive:true). However, I recently added fixes and unit tests for find and the function you are adding here does not expose the same options (and likely behaves differently too). We should drive consistency, not add different find/globbing functionality.

Comment (nigurr) [inline node/task.ts]: Can you point me to the tests which shows the existing pattern? 
I looked at from Task owners perspective, they expect to simply pass VSTS filters and expects the files to be discovered.

Comment (nigurr) [inline node/task.ts]: @ericsciple ping. We should close on this asap, as we have too much dependency on this functionality across different tasks :)

Comment (bryanmacfarlane) [inline node/task.ts]: We won't be able to this week.  Eric is OOF this week and I'm OOF till Monday.

Comment (ericsciple) [inline node/task.ts]: i still have the same questions and concerns

Comment (bryanmacfarlane) [inline node/task.ts]: Agreed on Eric's points above.

**Comment by bryanmacfarlane** [comment]
In addition to Eric's points, we need to also extend the mock class in the lib with the changes.

**Comment by ericsciple** [comment]
closing since `findMatch` has been added now which addresses other issues and `match` has been revamped

**Pull request closed without merging**